contracts/lifecycle/TimeFrame.sol

```diff
@@ -0,0 +1,71 @@
+pragma solidity ^0.5.2;
+
+/**
+ * @title Provides generic functionality for a TimeFrame.
+ * @author Sam Ruberti <sam@GRAFFT.io>
+ * @dev This library provides useful methods for governing a start and an end time.
+ *
+ * WARNING: This should NOT be used for contracts that depend on time for an outcome where a miner can benefit.
+ * One example would be a lottery where the first submission of an answer to a problem is rewarded. In this case the
+ * contract would be vulnerable to malicious miners who could read the answer of a not-yet-mined transaction then
+ * substitute their own answer in a new transaction before the victim's transaction, stealing the reward in the process.
+ *
+ * SAFE USE: TimeFrame CAN be used for approximate times that govern periods when methods and actions may occur.
+ * An example is an escrow that contains a start and expiration date. You can read about timestamp security
+ * vulnerabilities here:
+ * https://github.com/ethereumbook/ethereumbook/blob/first_edition_first_print/09smart-contracts-security.asciidoc#block-timestamp-manipulation
+ *
+ * IMPORTANT I emphasize that the time windows is not too narrow (e.g. a couple of seconds)
+ */
+
+
+/* solhint-disable not-rely-on-time*/
+library TimeFrame {
+    struct Epoch {
+        uint256 start;
+        uint256 end;
+    }
+
+    function hasStarted(Epoch storage _epoch) internal view returns (bool) {
```

Comment:
I think there's no need to call the argument `_epoch`, and we can simply rename to `epoch`.

Reply:
I'm pretty sure that was your idea in the first place but I changed it anyway

oh and by the way you mentioned me with wrong user name my user name is @startgeek not @catageek

Reply:
Oh my bad, I just typed 'geek' and GitHub's autocomplete tricked me :( Sorry both of you!